### PR TITLE
provide a unit test for the sdb utility

### PR DESCRIPTION
**Pull request by vernondcole**
### What does this PR do?

Provide a unit test for the \__utils\__['sdb.sdb_get'] and friends.

### What issues does this PR fix or reference?

Issue #43815 and PR #43600 

### Tests written?

It _is_ a test.

### CODE REVIEW

**Comment by vernondcole** [comment]
@garethgreenaway @rallytime @cachedout @gtmanfred I believe that this will justify that #43600 is not the cause of unit test failures.  If we get a red dot on ubuntu16-py3 then this is a valid test.

**Comment by garethgreenaway** [comment]
@vernondcole Can you update the dictionary created at line 60 to this:
`expected = {b'name': b'testone', b'number': 46}`

File: tests/unit/utils/test_sdb.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from tests.support.mock import (
+    NO_MOCK,
+    NO_MOCK_REASON
+)
+
+# Import Salt Libs
+import salt.utils.sdb as sdb
+import salt.exceptions
+
+TEMP_DATABASE_FILE = '/tmp/salt-tests-tmpdir/test_sdb.sqlite'
```
Comment (cachedout) [inline tests/unit/utils/test_sdb.py]: This will fail on Windows systems.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by cachedout** [comment]
Hi @vernondcole. It does looks like this correctly exposes the problem. Thank you!

As you know, your Vagrant PR is blocked on the resolution of this issue. Did you want to take a swing at doing the SDB fix as well?

File: tests/unit/utils/test_sdb.py (hunk 1 above)
Comment (vernondcole) [reply tests/unit/utils/test_sdb.py]: I had guessed that there is a global or an environment variable for this, but a scan of the entire code base using PyCharm did not find it.  I failed to guess the correct keywords, it seems.

Perhaps someone would be so kind as to tell me the secret?

Comment (gtmanfred) [reply tests/unit/utils/test_sdb.py]: You can use

```
from tests.support.runtests import RUNTIME_VARS
TEMP_DATABASE_FILE = os.path.join(RUNTIME_VARS.TMP, 'test_sdb.sqlite')
```

**Comment by vernondcole** [comment]
@cachedout Sorry, I am afraid I don't have time to find and fix the actual bug. I still have the vagrant state code, the vagrant cloud code, and a SaltConf presentation to finish, in addition to the work they pay me to do. I think writing the test is as far beyond the call of duty as I can push it. I have proven that the bug ain't mine.

Just be a good fellow and check in the Vagrant module.  Don't make me start out my SaltConf talk with "but you have to use my fork of Salt to do this ...".  It would sound way too lame.

File: tests/unit/utils/test_sdb.py (hunk 1 above)
Comment (vernondcole) [reply tests/unit/utils/test_sdb.py]: I like it.

**Comment by garethgreenaway** [comment]
re-run ubuntu16

**Comment by garethgreenaway** [comment]
re-run ubuntu

**Comment by garethgreenaway** [comment]
re-run py3

**Pull request merged**